From 76452744c01a68ff2c00747a556594fd4a6c64ff Mon Sep 17 00:00:00 2001
From: Gordon Messmer <gordon.messmer@gmail.com>
Date: Sun, 24 Nov 2019 15:26:39 -0800
Subject: [PATCH 2/5] scripts/pythondistdeps: Use rich deps for semantically
 versioned dependencies

(cherry picked from commit 403f75cdbb94ced52f169684c12ee5110e468dfa)
---
 scripts/pythondistdeps.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/scripts/pythondistdeps.py b/scripts/pythondistdeps.py
index 5897992ec..94707d8f4 100755
--- a/scripts/pythondistdeps.py
+++ b/scripts/pythondistdeps.py
@@ -242,11 +242,13 @@ names.sort()
 for name in names:
     if py_deps[name]:
         # Print out versioned provides, requires, recommends, conflicts
+        spec_list = []
         for spec in py_deps[name]:
             if spec[0] == '!=':
-                print('({n} < {v} or {n} >= {v}.0)'.format(n=name, v=spec[1]))
+                spec_list.append('{n} < {v} or {n} >= {v}.0'.format(n=name, v=spec[1]))
             else:
-                print('{} {} {}'.format(name, spec[0], spec[1]))
+                spec_list.append('{} {} {}'.format(name, spec[0], spec[1]))
+        print('(%s)' % ' with '.join(spec_list))
     else:
         # Print out unversioned provides, requires, recommends, conflicts
         print(name)
-- 
2.21.0

